searcher/searcher.go

```diff
@@ -287,7 +287,7 @@ func MakeAll(cfg *config.Config) (map[string]*Searcher, map[string]error, error)
 	lim := makeLimiter(cfg.MaxConcurrentIndexers)
 
 	// Channel to receive the results from newSearcherConcurrent function.
-	resultCh := make(chan searcherResult, 1)
+	resultCh := make(chan searcherResult)
```

Comment:
@dgryski IMO, making this channel unbuffered won't solve the problem of blocking writes from goroutines. How is unbuffered channel better than buffered channel of length 1?

Buffered channel of length `len(cfg.Repos)`, however, will make sense.

Reply:
Yes, I meant more that having a buffer size of 1 didn't make sense.  Either make it totally unbuffered (so they all block), or with enough space for them all to put the results.

Reply:
Ok, since we aren't doing any heavy work after receiving on the channel, I feel it's okay to make the goroutines blocking on sending to the channel. What do you think? Do you have a preference?

Reply:
Standard practice for launching a set of goroutines with a response channel is to have it buffered with the number of known entries so that none of them block.

Reply:
Ok, I'll use the standard practice then. Thanks 🙂